scripts/traiage.sh

```diff
@@ -0,0 +1,271 @@
+		create \
+		--template "${TEMPLATE_NAME}" \
+		--stop-after 30m \
+		--yes \
+		"${WORKSPACE_NAME}"
+	exit 0
+}
+
+prompt() {
+	if [[ -z "${AGENTAPI_SLUG}" ]]; then
+		prompt_ssh
+	else
+		prompt_agentapi
+	fi
+}
+
+ssh_config() {
+	requiredenvs CODER_URL CODER_SESSION_TOKEN WORKSPACE_NAME
+
+	if [[ -n "${OPENSSH_CONFIG_FILE:-}" ]]; then
+		echo "Using existing SSH config file: ${OPENSSH_CONFIG_FILE}"
+		return
+	fi
+
+	OPENSSH_CONFIG_FILE="${TEMPDIR}/coder-ssh.config"
+	"${CODER_BIN}" \
+		config-ssh \
+		--url "${CODER_URL}" \
+		--token "${CODER_SESSION_TOKEN}" \
+		--ssh-config-file="${OPENSSH_CONFIG_FILE}" \
+		--yes
+	export OPENSSH_CONFIG_FILE
+}
+
+prompt_ssh() {
+	requiredenvs CODER_URL CODER_SESSION_TOKEN WORKSPACE_NAME PROMPT
+
+	ssh_config
+
+	# Write prompt to a file in the workspace
+	cat <<<"${PROMPT}" >"${TEMPDIR}/prompt.txt"
+	scp \
+		-F "${OPENSSH_CONFIG_FILE}" \
+		"${TEMPDIR}/prompt.txt" \
+		"${WORKSPACE_NAME}.coder:/tmp/prompt.txt"
+
+	# Execute claude over SSH
+	# Note: use of cat to work around claude-code#7357
+	ssh \
+		-F "${OPENSSH_CONFIG_FILE}" \
+		"${WORKSPACE_NAME}.coder" \
+		-- \
+		"cat /tmp/prompt.txt | \"\${HOME}\"/.local/bin/claude --dangerously-skip-permissions --print --verbose --output-format=stream-json"
+	exit 0
+}
+
+prompt_agentapi() {
+	requiredenvs CODER_URL CODER_SESSION_TOKEN WORKSPACE_NAME AGENTAPI_SLUG PROMPT
+
+	wait
+
+	username=$(curl \
+		--fail \
+		--header "Coder-Session-Token: ${CODER_SESSION_TOKEN}" \
+		--location \
+		--show-error \
+		--silent \
+		"${CODER_URL}/api/v2/users/me" | jq -r '.username')
+
+	payload="{
+		\"content\": \"${PROMPT}\",
+		\"type\": \"user\"
+	}"
+
+	response=$(curl \
+		--data-raw "${payload}" \
+		--fail \
+		--header "Content-Type: application/json" \
+		--header "Coder-Session-Token: ${CODER_SESSION_TOKEN}" \
+		--location \
+		--request POST \
+		--show-error \
+		--silent \
+		"${CODER_URL}/@${username}/${WORKSPACE_NAME}/apps/${AGENTAPI_SLUG}/message" | jq -r '.ok')
+	if [[ "${response}" != "true" ]]; then
+		echo "Failed to send prompt"
+		exit 1
+	fi
+
+	wait
+}
+
+wait() {
+	requiredenvs CODER_URL CODER_SESSION_TOKEN WORKSPACE_NAME PROMPT
+	username=$(curl \
+		--fail \
+		--header "Coder-Session-Token: ${CODER_SESSION_TOKEN}" \
+		--location \
+		--show-error \
+		--silent \
+		"${CODER_URL}/api/v2/users/me" | jq -r '.username')
+
+	payload="{
+		\"content\": \"${PROMPT}\",
+		\"type\": \"user\"
+	}"
+
+	for attempt in {1..600}; do
+		response=$(curl \
+			--data-raw "${payload}" \
+			--fail \
+			--header "Content-Type: application/json" \
+			--header "Coder-Session-Token: ${CODER_SESSION_TOKEN}" \
+			--location \
+			--request GET \
+			--show-error \
+			--silent \
+			"${CODER_URL}/@${username}/${WORKSPACE_NAME}/apps/agentapi/status" | jq -r '.status')
+		if [[ "${response}" == "stable" ]]; then
+			echo "AgentAPI stable"
+			break
+		fi
+		echo "Waiting for AgentAPI to report stable status (attempt ${attempt}/600)"
+		sleep 1
```

Comment:
Suggestion: Slightly bumping this would reduce load and probably not affect responsiveness significantly? Maybe 3-5 sec would be good?